src/LodeRunner.API.Test/IntegrationTests/Controllers/TestRuns.cs

```diff
@@ -0,0 +1,156 @@
+            {
+                Assert.Equal(expectedValue, httpResponse.Content.Headers.ContentType.ToString());
+            }
+            else
+            {
+                Assert.True(false, $"Unable to process {nameof(this.CanGetEndpointsReturnSuccessAndCorrectContentType)} request.");
+            }
+        }
+
+        /// <summary>
+        /// Determines whether this instance [can get test runs].
+        /// </summary>
+        /// <returns><see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        [Trait("Category", "Integration")]
+        public async Task CanGetTestRuns()
+        {
+            using var httpClient = ComponentsFactory.CreateLodeRunnerAPIHttpClient(this.factory);
+
+            await httpClient.GetTestRuns(TestRunsUri, this.jsonOptions, this.output);
+        }
+
+        /// <summary>
+        /// Determines whether this instance [can post test runs].
+        /// </summary>
+        /// <returns><see cref="Task"/> representing the asynchronous integration test.</returns>
+        [Fact]
+        [Trait("Category", "Integration")]
+        public async Task CanCreateAndGetTestRunById()
+        {
+            using var httpClient = ComponentsFactory.CreateLodeRunnerAPIHttpClient(this.factory);
+
+            var postedTestRun = await httpClient.PostTestRun(TestRunsUri, this.jsonOptions, this.output);
+            var gettedTestRun = await httpClient.GetTestRunById(TestRunsUri + "/" + postedTestRun.Id, this.jsonOptions, this.output);
+
+            // Delete the TestRun created in this Integration Test scope
+            await httpClient.DeleteTestRunById(gettedTestRun.Id, TestRunsUri, this.output);
+
+            Assert.Equal(JsonSerializer.Serialize(postedTestRun), JsonSerializer.Serialize(gettedTestRun));
+        }
+
+        /// <summary>
+        /// Determines whether this instance [can get clients by identifier] the specified client status identifier.
+        /// </summary>
+        /// <returns><see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        [Trait("Category", "Integration")]
+        public async Task CanPutTestRuns()
+        {
+            using var httpClient = ComponentsFactory.CreateLodeRunnerAPIHttpClient(this.factory);
+
+            // Create a new TestRun
+            TestRun testRun = await httpClient.PostTestRun(TestRunsUri, this.jsonOptions, this.output);
+
+            Assert.NotNull(testRun);
+
+            var errorsCount = ParametersValidator<TestRun>.ValidateEntityId(testRun.Id).Count;
+
+            Assert.True(errorsCount == 0, $"Local Time:{DateTime.Now}\tUnable to Post a Sample TestRun item.");
+
+            // Generate a TestRunPayload from newly created TestRun
+            var testRunPayload = testRun.AutomapAndGetTestRunTestPayload();
+
+            // Update the existing TestRun
+            await httpClient.PutTestRun(testRunPayload, testRun.Id, TestRunsUri, this.jsonOptions, this.output);
```

Comment:
We should assert that the response is HTTP 204 on update (PUT).

Reply:
done